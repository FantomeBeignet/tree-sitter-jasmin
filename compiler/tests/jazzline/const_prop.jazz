u16[10] jzetas = {2285, 2571, 2970, 1812, 1493, 1422, 287, 202, 3158, 622};


export fn _test2 (reg ptr u16 [10] rp) -> (reg ptr u16 [10])
{
  inline int j;
  for j = 0 to 10
  {
    rp[(int)j] = jzetas[(int)j];
  }
  return rp;
}
