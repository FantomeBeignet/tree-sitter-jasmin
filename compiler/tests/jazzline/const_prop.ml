open Common

let () =
  parse_and_print "const_prop.jazz"  "_test2"
