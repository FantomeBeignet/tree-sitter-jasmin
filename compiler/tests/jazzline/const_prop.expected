proc _test2(rp_9_163@uint16, rp_8_162@uint16, rp_7_161@uint16,
            rp_6_160@uint16, rp_5_159@uint16, rp_4_158@uint16,
            rp_3_157@uint16, rp_2_156@uint16, rp_1_155@uint16,
            rp_0_154@uint16, rp_0_144@uint16, rp_1_145@uint16,
            rp_2_146@uint16, rp_3_147@uint16, rp_4_148@uint16,
            rp_5_149@uint16, rp_6_150@uint16, rp_7_151@uint16,
            rp_8_152@uint16, rp_9_153@uint16) =
{
 true && true
}
mov rp_0_154@uint16 (2285)@uint16;
mov rp_1_155@uint16 (2571)@uint16;
mov rp_2_156@uint16 (2970)@uint16;
mov rp_3_157@uint16 (1812)@uint16;
mov rp_4_158@uint16 (1493)@uint16;
mov rp_5_159@uint16 (1422)@uint16;
mov rp_6_160@uint16 (287)@uint16;
mov rp_7_161@uint16 (202)@uint16;
mov rp_8_162@uint16 (3158)@uint16;
mov rp_9_163@uint16 (622)@uint16;
{
 true && true
}
 